File: EndlessWorld/Assets/Collision HIT/Scripts JS/Game/TurnCamera.js
#pragma strict

var GeneralAngle:int;
var SpeedOfTurn:float;
var AudioPlay:AudioClip;
private var DontRepeat:boolean=false;

function OnTriggerEnter(other: Collider){ 
if(other.gameObject.tag=="Camera"){
if(!DontRepeat){ DontRepeat = true;
other.gameObject.GetComponent.<CameraMove>().RotateSpeed = SpeedOfTurn;
other.gameObject.GetComponent.<CameraMove>().RotateAngle = GeneralAngle;
other.gameObject.GetComponent.<CameraMove>().Rotate();
gameObject.GetComponent.<AudioSource>().PlayOneShot(AudioPlay);
gameObject.GetComponent.<BoxCollider>().enabled = false;
gameObject.transform.parent = GameObject.Find("StaticParent").transform;
}
}
}

File: EndlessWorld/Assets/Collision HIT/Scripts C#/Game/TurnCamera.cs
using UnityEngine;
using System.Collections;

public class TurnCamera : MonoBehaviour {

	public int GeneralAngle;
	public float SpeedOfTurn;
	public AudioClip AudioPlay;
	private bool DontRepeat = false;

	void OnTriggerEnter(Collider other){ 
		if(other.gameObject.tag=="Camera"){
			if(!DontRepeat){ DontRepeat = true;
				other.gameObject.GetComponent<CameraMove>().RotateSpeed = SpeedOfTurn;
				other.gameObject.GetComponent<CameraMove>().RotateAngle = GeneralAngle;
				other.gameObject.GetComponent<CameraMove>().StartCoroutine("Rotate");
				gameObject.GetComponent<AudioSource>().PlayOneShot(AudioPlay);
				gameObject.GetComponent<BoxCollider>().enabled = false;
				gameObject.transform.parent = GameObject.Find("StaticParent").transform;
			}
		}
	}

	void Update(){
		
	}

}
